train_model.py

```diff
@@ -331,15 +339,25 @@ def main():
     # Only init once, on rank 0 only
     if trainer.global_rank == 0:
         utils.init_wandb_metrics(logger)  # Do after wandb.init
-    if args.eval:
-        print_eval(args.eval)
-        if args.eval == "val":
-            data_module.split = "val"
-        else:  # Test
-            data_module.split = "test"
+
+    # Check if the mode is evaluation (either 'val' or 'test')
+    if args.eval in ["val", "test"]:
+        data_module.split = args.eval
         trainer.test(model=model, datamodule=data_module, ckpt_path=args.load)
+
+    # Check if the mode is prediction
+    elif args.eval == "predict":
+        data_module.split = "pred"
+        assert devices == 1, "Device not set to 1, check cuda availability"
+        # assert trainer.accelerator == "cuda", "Accelerator should be GPU"
```

Comment:
this fails if cuda devices > 1. you are requesting one node with 4 ntasks in your `slurm_predict.py` file so you will always get >1 device. Since the code should work on all devices (cpu, gpu) I suggest to remove these assertions for now. 
```suggestion
```